src/pybind/mgr/dashboard/controllers/smb.py

```diff
@@ -88,6 +89,26 @@
 }
 
 
+def raise_on_failure(func):
+    @wraps(func)
+    def wrapper(*args, **kwargs):
+        result = func(*args, **kwargs)
+
+        if 'success' in result and result['success'] is False:
```

Comment:
```suggestion
if result.get('success') is False:
```
What about this ? If the key does not exist, result.get('success') will return None, which won’t match False, so the condition will still work as expected.

Reply:
Sure, I've updated it.